apps/dashboard/src/components/activity/activity-filters.tsx

```diff
@@ -1,188 +1,105 @@
-import { ChannelTypeEnum } from '@novu/shared';
 import { CalendarIcon } from 'lucide-react';
-import { useEffect, useMemo } from 'react';
-import { useForm } from 'react-hook-form';
+import { ChannelTypeEnum } from '@novu/shared';
+
 import { useFetchWorkflows } from '../../hooks/use-fetch-workflows';
 import { Button } from '../primitives/button';
 import { FacetedFormFilter } from '../primitives/form/faceted-filter/facated-form-filter';
-import { Form, FormField, FormItem } from '../primitives/form/form';
+import { CHANNEL_OPTIONS, DATE_RANGE_OPTIONS } from './constants';
+import { ActivityFiltersData } from '@/types/activity';
+import { cn } from '@/utils/ui';
+
+type HideableFilter = 'dateRange' | 'workflows' | 'channels' | 'transactionId' | 'subscriberId';
 
 export type ActivityFilters = {
+  filters: ActivityFiltersData;
+  showReset?: boolean;
   onFiltersChange: (filters: ActivityFiltersData) => void;
-  initialValues: ActivityFiltersData;
   onReset?: () => void;
+  hide?: HideableFilter[];
+  className?: string;
 };
 
-export type ActivityFiltersData = {
-  dateRange: string;
-  channels: ChannelTypeEnum[];
-  workflows: string[];
-  transactionId: string;
-  subscriberId: string;
-};
-
-const DATE_RANGE_OPTIONS = [
-  { value: '24h', label: 'Last 24 hours' },
-  { value: '7d', label: 'Last 7 days' },
-  { value: '30d', label: 'Last 30 days' },
-];
-
-const CHANNEL_OPTIONS = [
-  { value: ChannelTypeEnum.SMS, label: 'SMS' },
-  { value: ChannelTypeEnum.EMAIL, label: 'Email' },
-  { value: ChannelTypeEnum.IN_APP, label: 'In-App' },
-  { value: ChannelTypeEnum.PUSH, label: 'Push' },
-  { value: ChannelTypeEnum.CHAT, label: 'Chat' },
-];
-
-export const defaultActivityFilters: ActivityFiltersData = {
-  dateRange: '30d',
-  channels: [],
-  workflows: [],
-  transactionId: '',
-  subscriberId: '',
-} as const;
-
-export function ActivityFilters({ onFiltersChange, initialValues, onReset }: ActivityFilters) {
-  const form = useForm<ActivityFiltersData>({
-    defaultValues: initialValues || defaultActivityFilters,
-  });
-
-  const watchedValues = form.watch();
-
-  const hasChanges = useMemo(() => {
-    return (
-      watchedValues.dateRange !== defaultActivityFilters.dateRange ||
-      watchedValues.channels.length > 0 ||
-      watchedValues.workflows.length > 0 ||
-      watchedValues.transactionId !== defaultActivityFilters.transactionId ||
-      watchedValues.subscriberId !== defaultActivityFilters.subscriberId
-    );
-  }, [watchedValues]);
-
+export function ActivityFilters({
```

Comment:
Refactored this component completely. First of all there was no need to use the form if it's not submitted or validated. Second thing that it was hacky and working only on the activity feed page as it uses the query params for the state. When I tried to reuse it for the Subscriber Activity List it wasn't working at all and had recurrent calls which was freezing the page.

So I removed the form and simplified the component a bit, now it's controlled component where the state is passed from the parent.